libcxx/include/tuple

```diff
@@ -1132,7 +1156,11 @@ struct __ignore_t
 };
 
 namespace {
+#ifdef __CUDA_ARCH__
+  _LIBCUDACXX_INLINE_VAR __constant__ __ignore_t<unsigned char> ignore = __ignore_t<unsigned char>();
+#else
   _LIBCUDACXX_INLINE_VAR constexpr __ignore_t<unsigned char> ignore = __ignore_t<unsigned char>();
+#endif
```

Comment:
A similar - though not identical - pattern also appeared in the complex changes; I wonder if there's a nice abstraction we could have for when we need to do selection of essentially a single token like this.

Reply:
I'll turn this into an issue. :)